### PR TITLE
Adding mongodb-url flag that can change the url of mongo, enabling to connect to a remote mongodb

### PR DESCRIPTION
**Pull request by liranmauda**
- Adding mongodb-url flag that can change the URL of mongo, enabling to connect to a remote mongodb
- Skipping create of db pod when given an external db URL

Expecting URL to start with:  `mongodb://` or `mongodb+srv://`

Fixes: #304

### CODE REVIEW

**Comment by liranmauda** [comment]
Moving to draft.
After talking with @lallinger-arbeit and @QuintinBecker we might want to drop this effort

**Comment by lallinger-arbeit** [comment]
@liranmauda could you do me a favour and rebase this? we are still using this and currently i am always rebasing this myself, but it gets harder for me to do it correctly with more and more changes...

**Comment by liranmauda** [comment]
> @liranmauda could you do me a favour and rebase this? we are still using this and currently i am always rebasing this myself, but it gets harder for me to do it correctly with more and more changes...

Are you using it? 
I will rebase.

**Comment by lallinger-arbeit** [comment]
> Are you using it?
yes, with the HA setup described [here](https://github.com/noobaa/noobaa-operator/issues/481)
> I will rebase.
thank you very much!

**Comment by liranmauda** [comment]
Done, Hopefully I didn't miss anything :) 
@lallinger-arbeit Can you test it?

**Comment by lallinger-arbeit** [comment]
Perfect, thanks! I will test it immediately!

**Comment by lallinger-arbeit** [comment]
@liranmauda seems to be not right yet, noobaa-core does not get the set mongoDbURL, but the endpoints do:
grafik
grafik

**Comment by lallinger-arbeit** [comment]
The attempts field now shows up as desired. Thinking about it i think it would be more straightforward if the field would be called `Retries` instead of `Attempts`. As attempts=0 would mean the trigger shouldn't be run... If you rename it to retries i would suggest the following tooltip: "Number of retries Noobaa will make if the trigger fails"

**Comment by liranmauda** [comment]
pushed a fix.
Changing to "ready for review"

File: pkg/system/phase2_creating.go [hunk 1]
```diff
@@ -87,8 +94,17 @@ func (r *Reconciler) ReconcilePhaseCreatingForMainClusters() error {
...
 		return err
 	}
-	if err := r.ReconcileDB(); err != nil {
-		return err
+	// create the mongo db only if mongo db url is not given.
+	if r.NooBaa.Spec.MongoDbURL == "" {
+		if err := r.UpgradeSplitDB(); err != nil {
+			return err
+		}
+		if err := r.ReconcileDB(); err != nil {
```
Comment (dannyzaken) [inline pkg/system/phase2_creating.go:102]: ReconcileDB also reconciles postgres. you should only refer to `MongoDbURL` when dealing with mongo. I think for now you can verify in phase1 that if `MongoDbURL` is passed then the `DB_TYPE` must be mongo. later on, we might need to handle the migration from external mongo to postgres, where the db-type is postgres and there is a mongoURL

**Review by dannyzaken — CHANGES_REQUESTED** [review]

**Comment by lallinger-arbeit** [comment]
> 
> 
> pushed a fix.

working as inteded now, thanks @liranmauda :)

**Review by dannyzaken — APPROVED** [review]

**Pull request merged**